pkgs/servers/web-apps/kavita/default.nix

```
@@ -0,0 +1,64 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildDotnetModule,
+  dotnetCorePackages,
+  buildNpmPackage,
+  autoPatchelfHook,
+}:
+buildDotnetModule rec {
+  pname = "kavita";
+  version = "0.7.1.4";
+
+  src = fetchFromGitHub {
+    owner = "kareadita";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-jNhiwyz6iVSLlvMNjI689TwQYuEvTJ+QaPvvDQ4UOwc=";
+  };
+
+  patches = [
+    # Disable swagger PostBuild, I couldn't get it to run
+    ./disable-swagger.diff
+    # So that we can set the webroot
+    ./change-webroot.diff
+  ];
+
+  frontend =  buildNpmPackage {
+    pname = "kavita-web";
+    inherit version src;
+
+    sourceRoot = "source/UI/Web";
+
+    npmFlags = [ "--legacy-peer-deps" ];
+    npmRebuildFlags = [ "--ignore-scripts" ]; # Prevent playwright from trying to install browsers
+    npmDepsHash = "sha256-w0CuTPyCQyAxULvqd6+GiZaPlO8fh4xLmbEnGA47pL8=";
+  };
+
+  postPatch = ''
+    sed -i 's:@WEB_ROOT@:$frontend/lib/node_modules/kavita-webui/dist:' API/Controllers/FallbackController.cs API/Startup.cs
```

```suggestion
    sed -i 's:@WEB_ROOT@:${frontend}/lib/node_modules/kavita-webui/dist:' API/Controllers/FallbackController.cs API/Startup.cs
```
$frontend is not expanded in `FallbackController.cs`
> `return PhysicalFile(Path.Combine("$frontend/lib/node_modules/kavita-webui/dist", "index.html"), "text/HTML");`

---

Oops, I forgot that single quotes prevent variables from expanding. I wonder if using `$frontend` (env var) or `${frontend}` is preferred.